ui/src/main/java/com/midtrans/sdk/ui/views/ewallet/xltunai/XlTunaiStatusActivity.java

```diff
@@ -0,0 +1,168 @@
+package com.midtrans.sdk.ui.views.ewallet.xltunai;
+
+import android.annotation.TargetApi;
+import android.content.ClipData;
+import android.content.ClipboardManager;
+import android.content.Context;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import com.midtrans.sdk.ui.R;
+import com.midtrans.sdk.ui.abtracts.BaseActivity;
+import com.midtrans.sdk.ui.abtracts.BasePaymentPresenter;
+import com.midtrans.sdk.ui.adapters.ItemDetailsAdapter;
+import com.midtrans.sdk.ui.constants.Theme;
+import com.midtrans.sdk.ui.widgets.DefaultTextView;
+import com.midtrans.sdk.ui.widgets.FancyButton;
+import com.squareup.picasso.Picasso;
+
+/**
+ * Created by ziahaqi on 4/10/17.
+ */
+
+public class XlTunaiStatusActivity extends BaseActivity {
+    public static final String EXTRA_MERCHANT_CODE = "merchant.code";
+    public static final String EXTRA_ORDER_ID = "order.id";
+    public static final String EXTRA_EXPIRATION = "expiration";
+    private static final String LABEL_ORDER_ID = "Order ID";
+    private static final String LABEL_MERCHANT_CODE = "Merchant Code";
+
+    private DefaultTextView textOrderId, textMerchantCode, textValidUntil;
+    private FancyButton buttonCopyOrderId, buttonCopyMerchantCode;
+    private DefaultTextView textTitle;
+    private RecyclerView recyclerItemDetails;
+    private FancyButton buttonFinish;
+
+    private BasePaymentPresenter presenter;
+    private ItemDetailsAdapter itemDetailsAdapter;
+
+    private String orderId, expirationTime, merchantCode;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_xltunai_status);
+        initProperties();
+        initViews();
+        initThemes();
+        setupViews();
+    }
+
+    private void initThemes() {
+        initThemeColor();
+        setBackgroundColor(buttonFinish, Theme.PRIMARY_COLOR);
+        setBackgroundColor(recyclerItemDetails, Theme.PRIMARY_COLOR);
+    }
+
+    private void initProperties() {
+        orderId = getIntent().getStringExtra(EXTRA_ORDER_ID);
+        merchantCode = getIntent().getStringExtra(EXTRA_MERCHANT_CODE);
+        expirationTime = getIntent().getStringExtra(EXTRA_EXPIRATION);
+        presenter = new BasePaymentPresenter();
+        itemDetailsAdapter = new ItemDetailsAdapter(new ItemDetailsAdapter.ItemDetailListener() {
+            @Override
+            public void onItemShown() {
+                // do nothing
+            }
+        }, presenter.createItemDetails(this));
+    }
+
+
+    private void setupViews() {
```

Comment:
Instead of using `setupViews` method name. We can divide the setup view functions into several specific view bind methods:

```java
private void initToolbar() {
    // Set title
    textTitle.setText(getString(R.string.xl_tunai));
    // Set merchant logo
    ImageView merchantLogo = (ImageView) findViewById(R.id.merchant_logo);
    Picasso.with(this)
        .load(presenter.getMerchantLogo())
        .into(merchantLogo);
    initToolbarBackButton();
}

private void initItemDetails() {
    // set item details
    recyclerItemDetails.setLayoutManager(new LinearLayoutManager(this));
    recyclerItemDetails.setAdapter(itemDetailsAdapter);
}

private void bindValues() {
    //set merchant data
    textMerchantCode.setText(merchantCode);
    textOrderId.setText(orderId);
    textValidUntil.setText(expirationTime);
}

private void initCopyButton() {
    buttonCopyOrderId.setOnClickListener(new View.OnClickListener() {
         @Override
         public void onClick(View view) {
             copyOrderId();
         }
    });

    buttonCopyMerchantCode.setOnClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View view) {
             copyMerchantCode();
        }
    });
}

private void initFinishButton() {
    buttonFinish.setOnClickListener(new View.OnClickListener() {
         @Override
         public void onClick(View view) {
            onBackPressed();
         }
    });
}
```